INSTALL.txt

```diff
@@ -0,0 +1,287 @@
+
+10. If one wants to build any domain's images by hand, at the time of development
+for instance, it is possible by going into desired directory and using poky to build:
+
+- For building Domain-0:
+cd workspace_base/build/build/tmp/work/x86_64-xt-linux/dom0-image-thin-initramfs/1.0-r0/repo/
+
+- For building DomD:
+cd workspace_base/build/build/tmp/work/x86_64-xt-linux/domd-image-weston/1.0-r0/repo/
+
+- For building DomA:
+cd workspace_base/build/build/tmp/work/x86_64-xt-linux/domu-image-android/1.0-r0/repo/
+
+source poky/oe-init-build-env
+
+- For building Domain-0:
+bitbake core-image-thin-initramfs
+
+- For building DomD or DomA:
+bitbake core-image-weston
+
+- For building DomA:
+bitbake android
+
+Usage:
+======
+
+Different helpers scripts and docs are located at:
+build-workspace/build/meta-xt-prod-devel/doc
+
+Let's consider available boot options in details.
+
+1. Using a storage device.
+In order to boot system using a storage device, required storage device should be prepared
+and flashed beforehand. The mk_sdcard_image_domu.sh script is intended to help with that:
+
+sudo ./mk_sdcard_image.sh -p /IMAGE_FOLDER -d /IMAGE_FILE
+
+Where, IMAGE_FOLDER is a path to a folder where artifacts live (in the context of this document
+it is a "deploy" directory) and IMAGE_FILE is an output image file or physical device how
+it is appears in the filesystem (/dev/sdx).
+
+1.1 In case of SDx card booting we just have to insert SD card to a Host machine and run a script,
+the latter will do all required actions automatically. All what we need to care about is to write
+proper environment variables from U-Boot command line (boot_dev/bootcmd) according to the chosen SDx.
+See workspace_base/meta-xt-prod-gen3-test/doc/u-boot-env-salvator-x.txt for details.
+
+1.2 In case of eMMC booting, we have to have an access to it in order to flash images.
+It is going to be not quite easy as for removable SD card, but the one of the possible ways is
+to prepare the image blob using the same script, copy resulting blob to NFS root directory,
+set system to boot via NFS, go to a DomD on target (where eMMC device is available)
+and using "dd" command just copy blob to eMMC.
+
+For example,
+
+prepare an image blob:
+sudo ./mk_sdcard_image.sh -p /IMAGE_FOLDER -d /home/emmc.img
+
+and then run on target:
+dd if=/home/emmc.img of=/dev/mmcblk0
+
+After getting eMMC flashed we have to choose it to be an boot device in a similar way as it is done
+for SD card.
+See u-boot-env.txt for details.
+
+So, as you can see, varying U-Boot's "boot_dev" and "bootcmd" environment variables
+and domain config's "extra" and "disk" options it is possible to choose  different boot device
+for each system component.
+
+Additional script available in the product.
+1. uirfs.sh.
+This script is intended to pack/unpack uInitramfs for Domain-0. It might be helpful since uInitramfs
+contains a lot of things which may changed during testing. The "xt" directory shipes all guest domain
```

Comment:
No 'e' in verb "ships". From wiktionary: "verb ship (third-person singular simple present ships)"